### DISCUSSION TITLE
Pocketbase Auth with Nuxt3

### DISCUSSION THREAD

**Discussion by thisisfel1x** [pocketbase/pocketbase#4538] [Q&A] [answered]
Hello everyone,
perhaps the question seems unnecessary, but I have not yet fully understood the implementation of Pocketbase as a Nuxt3 plugin. So I thought I would just ask here

In my current project I am using Pocketbase as a backend for the web application I am developing.
I copied the code from the README for Nuxt3 1 to 1, so to speak, and now use the methods of the SDK in the code via `useNuxtApp().$pb`. I do not (yet) use the `server/` directory, so I plan to solve all operations with the backend via the pocketbase plugin. This is where my first question arises: is security guaranteed? I use a normal HTTPS connection, so the data traffic is encrypted.

But now to my main question.
As a test, I set the `maxAge` of the cookie to 30 seconds to see what happens when the auth cookie expires.
As I understand it, `useNuxtApp().$pb.authStore.isValid` should actually return false when the auth cookie expires? In this case, however, true is returned every time and e.g. I can still list/create records. Only on a complete page reload, false is returned and I am redirected to the login page by the middleware (how it should be).

Is this only due to the implementation as a plugin? Or do I have a slightly bigger error in my code which I dont see? My actual goal would be that when the auth cookie expires, the user is automatically redirected to the login page without a complete refresh of the page (for safety/security reasons).

Thank you for any comments, tips or suggestions. I'm still partly intermediate, so please don't be too hard on me :)

Best regards

**Comment by ganigeorgiev** [comment, marked as answer]
> As I understand it, useNuxtApp().$pb.authStore.isValid should actually return false when the auth cookie expires?

No. Neither the JS SDK, nor the PocketBase server actually uses the Nuxt cookie.

The Nuxt cookie is used only as a mean to share the PocketBase auth state when sending requests between the browser and the Nuxt/node server.

Note that the Nuxt example in the README is usually intended for apps that relies on JS SSR, aka. your PocketBase requests flow in this case is usually either "node-proxy" (`browser <--> node <--> PocketBase`) or "mixed" (both the browser and node server are sending requests to PocketBase).

If you are not going to use the SDK in a JS SSR context then you don't need to create Nuxt plugin/middleware or even deal with cookies.

PocketBase was designed to be interact directly so it is safe to use the JS SDK in a browser only context (`browser <--> PocketBase`) as long as your collections API rules are properly configured (_this is also what I usually recommend and prefer to use_).
When running in a browser, the JS SDK uses `LocalStorage` to keep track of the auth state between page navigation.
If the JS SDK is used inside a browser (_or when used as admin in a node server env as shown in https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7621130_) then it is safe to have a global instance that you can use throughout your entire app.

---

_Some details how PocketBase authentication works:_
PocketBase authentication is fully stateless and there are no sessions stored on the server or session cookies.
A request is considered authenticated only if it has a valid `Authorization:TOKEN` sent with it.
When using the JS SDK this header is added automatically to each request if there is a `pb.authStore.token`.
`pb.authStore` represents a local state. `pb.authStore.isValid` only loosely checks whether the store has a loaded token and that its `exp` claim is still valid, but this doesn't guarantee that the token has correct signature and that it is coming from PocketBase.
The actual token verification is done by the PocketBase server when you send it with the request (_that is why in the Nuxt example there is an extra call to `authRefresh()` after the token from the cookie was loaded into the store_).

**Comment by celestialmk** [comment]
This is very nice to hear, coming from next to nuxt, I was about to use cookies, this makes things much simpler!!
